tests/proxyname_test.go

```diff
@@ -0,0 +1,165 @@
+package suites
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"github.com/k8sbykeshed/k8s-service-lb-validator/entities"
+	"github.com/k8sbykeshed/k8s-service-lb-validator/entities/kubernetes"
+	"github.com/k8sbykeshed/k8s-service-lb-validator/matrix"
+)
+
+func mustOrFatal(err error, t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// Pods switch to use specified set of clusterIPs,
+// either disabledClusterIPs or toggledClusterIPs for reachability access
+func podsUseClusterIPs(pods []*entities.Pod, ips []string) error {
+	if len(pods) != len(ips) {
+		return errors.New("pods number does not match the provided IP number")
+	}
+	for i, pod := range pods {
+		pod.SetClusterIP(ips[i])
+	}
+	return nil
+}
+
+// Services switch to use specified label, with key and value
+func servicesSetLabel(services kubernetes.Services, key, value string) error {
+	for _, service := range services {
+		if err := service.SetLabel(key, value); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Services switch to not having the label with key
+func servicesRemoveLabel(services kubernetes.Services, key string) error {
+	for _, service := range services {
+		if err := service.RemoveLabel(key); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func TestProxyName(t *testing.T) {
```

Comment:
Can we rename to something closer like `s/ProxyName/ProxyNamelabel`, so we kind of know this is related to a label and not a feature...

Reply:
yea sure!